### PR TITLE
drop mappable flag from DMA map

### PR DESCRIPTION
**Pull request by tmakatos**
The flags field belongs to VFIO and it's not a good idea to reuse as new
VFIO flags can break things. Instead, we derive whether or not a region
is mappable if a file descriptor is passed.

Signed-off-by: Thanos Makatos <thanos.makatos@nutanix.com>

### CODE REVIEW

File: lib/libvfio-user.c [hunk 1]
```diff
@@ -508,16 +508,16 @@ handle_dma_map(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg,
         dma_map->flags &= ~VFIO_USER_F_DMA_REGION_WRITE;
     }
 
-    if (dma_map->flags == VFIO_USER_F_DMA_REGION_MAPPABLE) {
+    if (dma_map->flags != 0) {
```
Comment (swapnili) [inline lib/libvfio-user.c:511]: I think we must be strict and fail if client passes unknown flags?

**Review by swapnili — CHANGES_REQUESTED** [review]
VFIO_USER_F_DMA_REGION_MAPPABLE is also mentioned in include/libvfio-user.h and also used in test/py/
Can you also fix them?

**Comment by tmakatos** [comment]
> VFIO_USER_F_DMA_REGION_MAPPABLE is also mentioned in include/libvfio-user.h and also used in test/py/
> Can you also fix them?

Sure.

File: lib/libvfio-user.c (hunk 1 above)
Comment (tmakatos) [reply lib/libvfio-user.c:511]: Ah I forgot the `return` statement.

**Comment by tmakatos** [comment]
@swapnili addressed your comments, please take another look.

**Review by swapnili — APPROVED** [review]

**Comment by tmakatos** [comment]
Since this is a spec change I'll wait for @jlevon to double check.

**Review by jlevon — APPROVED** [review]

**Pull request merged**